libeos-payg/efi.c

```diff
@@ -537,26 +544,60 @@ efivarfs_read (const char  *name,
 
 /* eospayg_efi_secureboot_active:
  *
- * Check if the system was booted via SecureBoot
+ * Check if the system was booted via real SecureBoot in result
  *
  * Returns: %TRUE if booted via SecureBoot, %FALSE otherwise
  */
 gboolean
 eospayg_efi_secureboot_active (void)
 {
-  g_autofree char *tname = full_efi_name (GLOBAL_VARIABLE_GUID, "SecureBoot");
-  g_autofree unsigned char *content = NULL;
-  int size;
+  g_return_val_if_fail (efi != NULL, FALSE);
 
-  /* In test mode let's pretend secure boot is enabled */
-  if (test_mode)
-    return TRUE;
+  g_autofree char *name = full_efi_name (GLOBAL_VARIABLE_GUID, "SecureBoot");
 
-  content = efivarfs_read (tname, &size, NULL);
-  if (!content || size != 1)
-    return FALSE;
+  return eospayg_efi_var_read_fullname_boolean (name, NULL);
```

Comment:
This changes the behaviour of this function in tests. Previously in the test suite which calls `eospayg_efi_init (EOSPAYG_EFI_TEST_MODE, ...)` this function would unconditionally return TRUE. Now it will return FALSE.

Reply:
fixed in the new commit